js/src/sdk/base.toolset.ts

```diff
@@ -219,23 +219,32 @@ export class ComposioToolSet {
         connectedAccountId?: string,
     ): Promise<Record<string, any>> {
         // Custom actions are always executed in the host/local environment for JS SDK
-        if(await this.isCustomAction(action)) {
+        if (await this.isCustomAction(action)) {
+            let accountId = connectedAccountId;
+            if (!connectedAccountId) {
+                // fetch connected account id
+                const connectedAccounts = await this.client.connectedAccounts.list({
+                    user_uuid: entityId
+                });
+                accountId = connectedAccounts.items[0].id;
```

Comment:
Consider adding error handling for cases where no connected accounts are found. Currently, the code assumes the first account exists and tries to access it directly which could lead to runtime errors.